feat(content): Successors: Ghosts I (#12221)
endless-sky/endless-sky: data/successors/successor 2 ghosts aqrabe.txt

# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
# PARTICULAR PURPOSE. See the GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along with
# this program. If not, see <https://www.gnu.org/licenses/>.

mission "Successors: Ghosts Aqrabe 1"
	landing
	name "Keystone delivery"
	description "House Aqrabe wants 50 quantum keystones at <destination>."
	source "Kasii-Cavasaa-Oa"
	destination "Kasii-Cavasaa-Oa"
	to offer
		or
			and
				has "successors: ghosts stage 1"
				random < 15
			has "Successors: Ghosts 4: offered"
	on offer
		conversation
			`Shortly after landing here, you are approached by a tall Successor in unusual dress; though dark, it shimmers in the light with many little golden flecks. It is unlike any other Successor outfit you've seen. "Greetings, Captain," they say, curiously sonorous. "Noble House Aqrabe kindly requests thy presence in the eastern spaceport reliquary. I've been instructed to take thee there, at thy own discretion, of course."`
			choice
				`	"Very well."`
					goto reliquary
				`	"No, thank you."`
					decline
			label reliquary

> 			`	They lead you through several restricted areas of the spaceport to a thin tower on its periphery. The room you enter into is tremendous and circular; a shallow watery depression in the center is matched by an enormous semicircular tank that bulges down from above, almost touching the floor as it swoops towards the wet floor. What's inside it proves even more striking, however: soon, what look like the severed limbs of Successors begin to writhe downwards in the tank, each suspended by a number of glittering golden threads that snake up into the dark abyss of the overhead water. You look around to ask your guide what's happening, but they have already left. One of the arms rubs up against the glass, its undead suckers making faint popping sounds as they pass over it.`

Reviewer: ```suggestion
			`	They lead you through several restricted areas of the spaceport to a thin tower on its periphery. The room you enter into is tremendous and circular; a shallow watery depression in the center is matched by an enormous semicircular tank that bulges down from above, almost touching the wet floor as it swoops towards it. What's inside it proves even more striking, however: soon, what look like the severed limbs of Successors begin to writhe downwards in the tank, each suspended by a number of glittering golden threads that snake up into the dark abyss of the overhead water. You look around to ask your guide what's happening, but they have already left. One of the arms rubs up against the glass, its undead suckers making faint popping sounds as they pass over it.`
```

				accept
	to complete
		or
			has "Successors: Ghosts Aqrabe Remnant: offered"
			has "Successors: Ghosts Aqrabe Hai: offered"
	on complete
		log "Delivered a shipment of quantum keystones to House Aqrabe for study and installation on their ships."

conversation "successor keystone delivery"
	`(You have collected the fifty keystones requested by House Aqrabe. Would you like to turn them in?)`
	choice
		`	(Yes.)`
		`	(Not right now.)`
			defer
	`	You ask around at the spaceport, and you are soon directed to a side outfitting bay that seems to be the domain of some of House Aqrabe's more unorthodox efforts - the walls are draped with strange-looking fabrics, and a couple of unusual, abstract statues dot the facility. In the center, a ship-mounted scanner is in the midst of being deconstructed; as you walk in, a Successor pops their head around from behind it.`
	`	"Captain <last>?" they ask.`
	`	"Yes?"`
	`	They slither forward towards you and splay their arms in the traditional gesture. "Tiliia," they offer as an introduction. "I was told to expect thine arrival. Thou'st brought," and they pause to pronounce the word, "keystones, yes?"`

> 	`	"I have," you say. "Fifty tons, as was requested." Tiliia joins you in walking back to the <ship> to examine the cargo; when they're sufficiently satisfied that you've brought the proper amount, they pull out a data-pad and enter a couple of values. Though their eyes widen when you provide them with your receipts, they nonetheless confirm a few moments later that House Aqrabe has credited you to offset the purchase price of the keystones, plus a three hundred thousand-credit bonus.`

Reviewer 2: > Fifty tons

Five tons or fifty pieces, actually. Each Keystone weighs 0.1 tons

	`	As they begin directing ground crews to offload the keystones, Tiliia turns to you for a moment and requests that you meet them back at the spaceport by the end of the day: "thine assistance might be a boon to us in the next part of this journey," they say.`
		decline

mission "Successors: Ghosts Aqrabe Remnant"
	landing
	source "Kasii-Cavasaa-Oa"
	to fail
		has "Successors: Ghosts Aqrabe Hai: done"
	to offer
		has "Successors: Ghosts Aqrabe 1: offered"
	on offer
		conversation "successor keystone delivery"
		outfit "Quantum Key Stone" -50
		payment 12300000

mission "Successors: Ghosts Aqrabe Hai"
	landing
	source "Kasii-Cavasaa-Oa"
	to fail
		has "Successors: Ghosts Aqrabe Remnant: done"
	to offer
		has "Successors: Ghosts Aqrabe 1: offered"
	on offer
		conversation "successor keystone delivery"
		outfit "Quantum Keystone" -50

> 		payment 900000

Reviewer: Not sure if I'm missing something here, but in addition to the above, how does the game tell whether you got your keystones from the Remnant or the Hai? Haven't playtested this yet, so I might be missing some context.

> 		payment 900000

Author: There are separate missions that check whether the player has 50 of the Remnant keystone outfits, or 50 of the Hai ones. I'm not sure it's possible to require an undefined mixture of the two (50 total), so the player's only options are 50 of a single variety.

The outstanding problem, which I not have yet been able to resolve through my own tests, is that the player has to take off and reland before the next mission will be offered. I think I'll have to change the structure of these under the hood to fix that.

mission "Successors: Ghosts Aqrabe 3"
	name "Visit stellar garden"
	description "Attempt to land on the star of the <system> system."
	source "Kasii-Cavasaa-Oa"
	stopover "Exotic Metal Garden"
	destination "Kasii-Cavasaa-Oa"
	passengers 1
	to offer
		has	"Successors: Ghosts Aqrabe 2: done"
	on offer
		event "successors: ghosts star unrestrict"
		conversation
			`You meet Tiliia at their workshop, where they wave you inside. The scanner on their table has now been replaced by a collection of quantum keystones suspended in a variety of machines that prod and scan the stones incessantly. "Good news for thee, Captain, for all of us," Tiliia says. "The data from our expedition hath revealed some measure of the mystery of thy stones. They're oscillators of a sort, ones that sing in tune with the wavers of the wormhole..." Tiliia stops for a moment. "But we will have time to discuss that later. For now, what thou needest know is this: with the right materials, we may be able to magnify their effect. Hast thou visited the system Kasii-Cavaasa before, within the Tangles?"`
			choice
				`	"I have."`
				`	"Where are you going with this?"`

> 			`	Tiliia continues. "The star of that system is borne upon by a number of stellar metal gardens, or at least 'twas fifteen hundred years ago. Their bounty is here the key to our success. Forty tons or so would be enough to subdivide those keystones to supply all of the New Houses. We could render our wormhole-crossing armor obsolete - improve our probes, perform more archaeology." They pause, and a wave of color bounds down their limbs. "It could even help solve this present mystery of ours. So what thinks thou?"`

Reviewer: Might be worth it to provide more clarification on what Tiliia means by stellar metal gardens?

				flee
	on complete
		log "Attempted to retrieve stellar metal from Predecessor space for House Aqrabe, but was driven back by the intense heat."
		conversation
			`Tiliia has spent most of the journey holed away, poring over data gathered from your stardiving expedition, but it's not long after the <ship>'s landing gear touch the ground of the <planet> spaceport that they are out and about, stretching their arms in the air. They remain nearby until you've finished post-flight checks and thank you for transporting them, now surrounded by several scholarly-dressed Successors. Before they head off, they ask you to return to the spaceport by sundown.`
			`	Shortly before you depart into the port yourself, you overhear some of the ground crew remarking on the <ship>'s scorched hull.`

mission "Successors: Ghosts Aqrabe 4"
	name "Pick up ship from Kaatrij"
	description "Pick up a modified Vujlet from House Kaatrij at <destination>."
	clearance
	source "Kasii-Cavasaa-Oa"
	destination "Raaqa-Puan-Uuoru"
	to offer
		has	"Successors: Ghosts Aqrabe 3: done"
	on offer
		conversation

> 			`Tiliia stands waiting for you in the same statue garden they met you earlier, though this time they are alone, tapping at a metallic pad under an overhanging light. They offer a curt greeting of colors when they see you. "I've news both foul and fair for thee," they say. "Thy craft shall not suit for our task. Perhaps with a year of study it could be adapted to our aims, but Aqrabe and I agree that would not do."`

Reviewer: ```suggestion
			`Tiliia stands waiting for you in the same statue garden where they met you earlier, though this time they are alone, tapping at a metallic pad under an overhanging light. They offer a curt greeting of colors when they see you. "I've news both foul and fair for thee," they say. "Thy craft shall not suit for our task. Perhaps with a year of study it could be adapted to our aims, but Aqrabe and I agree that would not do."`
```

			`	This I myself attest as K- I- Saajret, of Somber Sejra."`
			label introduction
			`	A single Successor emerges from the side of the craft; their outfit does not resemble that of an archivist so much as that of a soldier, with armor-woven fabrics and a kinetic rifle held at their side. "Captain <first> <last>?" they say, and when you confirm, continue, "I shall accompany thee as an observer."`
			choice
				`	"I was told to expect an archivist."`
				`	"Very well."`
					goto end
			`	"Thine expectation is no longer accurate," they reply, not belying any emotion.`
			label end
			`	They show you onboard. The interior of Successor military ships looks much different from their civilian craft - the bridge is buried deep in the center of the craft under layers of armor, a small, circular room with a water-filled pool in the center and liquid-metal displays ringing the edges. Similarly compact living quarters and operational spaces fan out from it, each one well-equipped.`

> 			`	After you've had a look around, Kaatrij's observer esorts you off the vessel. "I and a Kaatrij crew shall pilot the vessel to Aqrabe. Once there, thou shoulds't install thy jump drive aboard."`

Reviewer 3: ```suggestion
			`	After you've had a look around, Kaatrij's observer esorts you off the vessel. "I and a Kaatrij crew shall pilot the vessel to Aqrabe. Once there, thou shouldst install thy jump drive aboard."`
```

				accept
	on visit
		dialog "You have returned to Kasii-Cavasaa-Oa, but the Vujlet you are escorting has not yet arrived."
	on complete
		give ship "Vujlet (Stellar)" "Nnesa ti-a-Oj"
		conversation
			`Both your ship and the Vujlet are directed to an enormous outfitting bay at <planet>'s spaceport, easily large enough for both craft ten times over. As you disembark, you notice several Successors are entering the landed Vujlet with large carts of equipment in tow, much of which is locked away in unmarked metallic cases. Notably, Kaatrij's representative remains onboard, keeping their distance from the Aqrabe ground crews. Tiliia spots you from an overhanging balcony and waves you over.`
			`	"A lovely craft, is it not, Captain? Thou hast not flown one of our martial craft before, hast thou?."`

> 			choice

Reviewer: Minor thing but it would be nice to have a choice where you can tell Tiliia that you found the ship fascinating, or something along those lines. All of the options right now seem kind of blasé?

				`	"A ship is a ship, really."`
				`	"It's not my first Successor craft, you know."`
					to display
						or
							has "ship model (all): Ejniisde"
							has "ship model (all): Kvelqde"
							has "ship model (all): Kvelqde-Khora"
							has "ship model (all): Myaede"
							has "ship model (all): Nnaatej"
							has "ship model (all): Nnesaarau"
							has "ship model (all): Ojde"
							has "ship model (all): Stolsaqra"
					goto experienced
			`	Tiliia makes a short humming sound. "Perhaps," they say. "Thou shalt have the opportunity to decide for thyself soon enough."`
				goto next
			label experienced

> 			`	Tiliia ripples with an emotion that's difficult to place, one tinted with shades of pride.`

Reviewer: What does 'shades of pride' mean?

			choice
				`	"I can understand wanting to make a difference."`
					goto difference
				`	"Do you really think the scions of the House will be that grateful?"`
			`	"'Tisn't about gratitude, nor about the scions. 'Tis a matter of... continuity. Our lives are waves upon the bright shore, shaped by predecessors, shaping consequents. Over the eons, names become words in the common language, and are then chosen again as names as new children become adults. In a thousand years, thy name may become the word for those 'keystones' of thine, and in a thousand more, it could be invoked by a new cohort in a new world."`
			choice
				`	"That seems a little far-fetched."`
					goto far-fetched
				`	"I wouldn't mind that."`
					goto egoist
				`	"Sometimes I worry they'll be invoking it as a curse. The Successors haven't done right by everyone, after all."`
					to display
						has "Successors: People's Houses: First Contact: offered"
			action
				set "successors confronted tiliia"
			`	Tiliia looks down at you with interest. "Perhaps I guess what thou mean'st, but tell me more, if it pleases thee."`

> 			`	"Well," you say, "the People's Houses don't seem happy with their lot, nor do the Predecessor planets I've seen look very comfortable. Not that humans don't have our own problems, but it seems like even thousands of years of 'waves upon the bright shore' haven't made the Successors totally enlightened. Supplying you with keystones might help us both right now, but who knows what'll come of it in the future?"`

Reviewer: ```suggestion
			`	"Well," you say, "the People's Houses don't seem happy with their lot, and the Predecessor planets that I've seen don't look very comfortable. Not that humans don't have our own problems, but it seems like even thousands of years of 'waves upon the bright shore' haven't made the Successors totally enlightened. Supplying you with keystones might help us both right now, but who knows what'll come of it in the future?"`
```

				goto end
			label difference
			`	"It's quite a feeling," they say, "isn't it?" They pause for a short while, looking up at the sky. The clouds overhead are starting to part and the snowfall is thinning. You notice for the first time that the ground is slightly heated - none of the snow that's falling has accumulated, instead just melting away into tiny channels in the rock. "My thanks to thee for joining me, captain," says Tiliia. "Thou shouldst prepare now - join us back at the spaceport when thou'rt ready."`
				goto end
			label alone
			`	"Of course. Join me back here in the spaceport, then, four hours hence."`
			label end
			`	With that, Tiliia heads off. You should make sure that the Vujlet you have been loaned is equipped for the upcoming voyage.`

mission "Successors: Ghosts Aqrabe 6"
	name "Visit exotic metal garden again"
	description "Attempt a landing on the <system> stellar metal garden using the modified Vujlet."
	source "Kasii-Cavasaa-Oa"
	destination "Exotic Metal Garden"
	passengers 2
	to offer

> 		has	"Successors: Ghosts Aqrabe 5: done"

Reviewer 4: ```suggestion
		has	"Successors: Ghosts Aqrabe 5: done"
		has "cargo space free" 40
	blocked
		"You need to have 40 tons of cargo space free to bring back enough stellar metal from <destination>."```

This would fix vgaming's issue, I think?

	on offer
		conversation
			`You meet Tiliia and House Kaatrij's observer back in the hangar bay with the specially-equipped Vujlet, which now gleams all over with an oily, iridescent sheen. Tiliia and the observer are in the middle of a heated conversation; you catch a bit of something ending "...foolish to trust an alien," from the observer.`
			`	Tiliia waves you over. "Good, captain. Our ally was just expressing the importance of thy promptness in this task. Kaatrij longs not to be parted from their trinket for long."`
			`	"Fine," you say. "Anything else I need to know?"`
			`	The observer looks you over with an emotionless gaze. "Our efforts shall protect the craft from the star's heat, but little else. 'T'would be wise not to linger in the Tangles."`
			`	Tiliia rubs their arms together and a pulse of electric blue passes over them. "Time to go, then, captain!"`
				accept

mission "Successors: Ghosts Aqrabe 7"
	name "Return to Kasii-Cavasaa-Oa"
	description "Escape the drone ship and return to <destination> with the stellar metal."
	landing
	source "Exotic Metal Garden"
	destination "Kasii-Cavasaa-Oa"
	cargo "Stellar metal" 40
	passengers 2
	to offer

> 		has	"Successors: Ghosts Aqrabe 6: done"

Reviewer 4: ```suggestion
		has	"Successors: Ghosts Aqrabe 6: done"
	blocked
		"You will need <capacity> to collect the stellar metal. Return in the modified Vujlet when you have the space free."```

	on offer
		conversation

> 			`Several minutes pass after landing before the hull of the <ship> has cooled enough to make it safe for exit - despite Kaatrij's loaned technology, the ship's armor still glows red-hot. The interior of the garden's loading bay is cavernous, the tangled metal strands drawn inside onto giant spools each many meters in diameter.`

Reviewer 2: Oof, it's so easy to miss this mission! I had to dig into the source code to figure out that me uninstalling the Cargo Extensions was the problem.

I think it's not intended to be "hidden", right? In this case, maybe it makes sense to have some helper mechanism to nudge the player into having 40 "tons" of cargo space available?

> 			`	Your crew gets to work under the supervision of Tiliia and Kaatrij's observer, removing small bits of metal from the rolls and carting them, one by one, onto the ship. Soon, Tiliia calls you over to a side section of the facility, and points out one of spools - unlike the others, this one is entirely empty.`

Reviewer: ```suggestion
			`	Your crew gets to work under the supervision of Tiliia and Kaatrij's observer, removing small bits of metal from the rolls and carting them, one by one, onto the ship. Soon, Tiliia calls you over to a side section of the facility, and points out one of the spools - unlike the others, this one is entirely empty.`
```

			`	"Do you know something about those ships?" you ask them, eliciting a long groan and a wave of oily black over their skin.`
			`	"This is what we earn for tearing up a nest; going where we shouldn't," they say. "'Twas a drone, an automated guardian from the days that garden was new. No doubt meant to chase off interlopers, as ourselves. But-"`
			choice
				`	"What is it?"`
				`	"Spit it out!"`
			`	"Such a thing should not be. No House would've left a craft like that behind to molder, not for the sake of some petty metal. Heirlooms of that age are precious - I cannot see why such a thing should be so eas'ly discarded." They finally relax a bit, and the black wave subsides.`
			`	"Thou hast done us a duty, nonetheless. Thou shalt see thine account flusher soon."`
			choice
				`	"What happens now?"`
				`	"What about this ship? Do I get to keep it?"`
					goto ship
			label now

> 			`	"Now we put this bounty to work, captain. I and the others here have a long task before us: I believe we shall be able to split up the keystones and amplify their effects with the stellar metal - tis only a matter of how much and how well. There's little thou can do to assist now, though I'll be sure to note thy helpfulness to Aqrabe in this task."`

Reviewer 3: ```suggestion
			`	"Now we put this bounty to work, captain. I and the others here have a long task before us: I believe we shall be able to split up the keystones and amplify their effects with the stellar metal - 'tis only a matter of how much and how well. There's little thou canst do to assist now, though I'll be sure to note thy helpfulness to Aqrabe in this task."`
```

			choice
				`	"What about this ship? Do I get to keep it?`
			label ship

> 			`	Tiliia nods. "Before we left, I heard from our fleet liaison with Kaatrij. The ship lacks weapons or anything too perilous for thou to keep, and Kaatrij has no doubt already started removing their device. Thou might'st consider it a bonus to thy fee, a favor from the House to thee."`

Reviewer 3: ```suggestion
			`	Tiliia nods. "Before we left, I heard from our fleet liaison with Kaatrij. The ship lacks weapons or anything too perilous for thee to keep, and Kaatrij has no doubt already started removing their device. Thou might'st consider it a bonus to thy fee, a favor from the House to thee."`
```
